src/wallet/wallet.cpp

```diff
@@ -4567,7 +4567,7 @@ ScriptPubKeyMan* CWallet::AddWalletDescriptor(WalletDescriptor& desc, const Flat
         }
 
         CTxDestination dest;
-        if (ExtractDestination(script_pub_keys.at(0), dest)) {
+        if (!internal && ExtractDestination(script_pub_keys.at(0), dest)) {
```

Comment:
Is it expected that we will set an empty label even when a label is not specified in request at all?

nit: I think it's more clear to make all the checks in the parent `if` (for range, internal and for label emptiness). This also might help to avoid unnecessary log message above.

Reply:
> Is it expected that we will set an empty label even when a label is not specified in request at all?

Yes.

Reply:
Moved it up.

Reply:
On second thought, nevermind. Reverted back to previous.

Moving it up will make the boolean logic at the outer if a little more confusing, and I think it's fine (perhaps even good) to be checking that scriptPubKeys were being generated for non-ranged things.

Reply:
> On second thought, nevermind. Reverted back to previous.
>
> Moving it up will make the boolean logic at the outer if a little more confusing, and I think it's fine (perhaps even good) to be checking that scriptPubKeys were being generated for non-ranged things.

Though the check becomes redundant after #20153 I think I maybe even should drop it in my PR